qpc/utils.py

```diff
@@ -356,7 +358,23 @@ def write_file(filename, content):
     """
     result = None
     input_path = os.path.expanduser(os.path.expandvars(filename))
-
-    with open(input_path, 'w') as out_file:
+    mode = 'w'
+    if binary:
+        mode = 'wb'
+    with open(input_path, mode) as out_file:
         out_file.write(content)
     return result
+
+
+def extract_json_from_tar(fileobj_content):
+    """Extract json data from tar.gz bytes.
+
+    :param fileobj_content: BytesIo object with tarball of json dict
+    """
+    filename = '/tmp/cli_tmp_%s.tar.gz' % time.strftime('%Y%m%d_%H%M%S')
+    write_file(filename, fileobj_content, True)
```

Comment:
Do you delete this file later?  I will be running this to verify, but it appears to leave a tar.gz behind.